### PR TITLE
Remove custom request id and use header propagation

### PR DESCRIPTION
**Pull request by DmyMi**
Propagate request tracing from inbound proxy to webapi to authserver.
Current http context enricher extracts `Request-Id` header automatically. Some other implementations use `X-Request-Id` so we propagate both to http client.

### CODE REVIEW

File: OutOfSchool/OutOfSchool.AuthCommon/Services/CommonMinistryAdminService.cs [hunk 1]
```diff
@@ -90,9 +88,9 @@ async Task<ResponseDto> CreateMinistryAdminOperation()
                     await transaction.RollbackAsync();
 
                     logger.LogError(
-                        $"Error happened while creation MinistryAdmin. Request(id): {requestId}" +
-                        $"User(id): {userId}" +
-                        $"{string.Join(Environment.NewLine, result.Errors.Select(e => e.Description))}");
+                        "Error happened while creation MinistryAdmin. User(id): {UserId}. {Error}",
+                        userId,
+                        string.Join(Environment.NewLine, result.Errors.Select(e => e.Description)));
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.AuthCommon/Services/CommonMinistryAdminService.cs]: (Code style) The next code pattern 
```
string.Join(Environment.NewLine, result.Errors.Select(e => e.Description)))
```
seems to be duplicated in subsequent calls. Couldn't it be seprated as an extension method?

Comment (VadymLevkovskyi) [reply OutOfSchool/OutOfSchool.AuthCommon/Services/CommonMinistryAdminService.cs]: Also same thing found in `ProviderAdminService`

File: OutOfSchool/OutOfSchool.WebApi/Startup.cs [hunk 2]
```diff
@@ -392,5 +395,11 @@ public static void AddApplicationServices(this WebApplicationBuilder builder)
             .AddDbContextCheck<OutOfSchoolDbContext>(
                 "Database",
                 tags: new[] { "readiness" });
+
+        services.AddHeaderPropagation(options =>
+        {
+            options.Headers.Add("Request-Id");
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.WebApi/Startup.cs]: (Logic) Don't you want to add some default value builder? (As a result for case when `Request-Id` is empty on one step you could rehidrate it by some GUID and track all subsequent steps)

**Review by VadymLevkovskyi — APPROVED** [review]
Aproved with minor suggestions (though they may be rejected or done in separate PR)

**Comment by DmyMi** [comment]
@VadymLevkovskyi thank you, fixed comments in new commit.

**Comment by VadymLevkovskyi** [comment]
> @VadymLevkovskyi thank you, fixed comments in new commit.

Not at all @DmyMi, changes look nice!

**Pull request merged**